router/relay/relay.go

```diff
@@ -1803,3 +1818,63 @@ func (rst *RelayStateImpl) ProcessMessage(
 func (rst *RelayStateImpl) ConnMgr() poolmgr.PoolMgr {
 	return rst.manager
 }
+
+func (rst *RelayStateImpl) loadColumns(schemaName, tableName string) ([]string, error) {
```

Comment:
We should definitely use connection different from user one. The design i have in mind is separate component (cache, syscache or maybe schemaCache) with its own pool. which is shared among all RelayState's.